charts/internal/shoot-storageclasses/templates/storageclasses.yaml

```diff
@@ -1,24 +1,30 @@
-{{- if not .Values.useLegacyProvisioner }}
+{{- $storageVersion := (include "storageclassversion" .) }}
+{{- range $key, $value := .Values.storageclasses }}
 ---
-apiVersion: {{ include "storageclassversion" . }}
+apiVersion: {{ $storageVersion }}
 kind: StorageClass
 metadata:
-  name: default-class
+  name: {{ $value.name }}
   annotations:
-    storageclass.kubernetes.io/is-default-class: "false"
     resources.gardener.cloud/delete-on-invalid-update: "true"
-allowVolumeExpansion: true
-provisioner: cinder.csi.openstack.org
-volumeBindingMode: WaitForFirstConsumer
----
-apiVersion: {{ include "storageclassversion" . }}
-kind: StorageClass
-metadata:
-  name: default
-  annotations:
+    {{- if $value.default }}
     storageclass.kubernetes.io/is-default-class: "true"
-    resources.gardener.cloud/delete-on-invalid-update: "true"
+    {{- else }}
+    storageclass.kubernetes.io/is-default-class: "false"
```

Comment:
Can we not simply omit this block in case the storageclass shouldn't be default?

Reply:
Yes this should be possible. I think it is more like a styling topic. I like the visibility of the default state. But I can remove it if you like.